### PR TITLE
fix-gateway-svc-annotations

### PR DESCRIPTION
**Pull request by steevi**
**Please provide a description of this PR:**
The annotations added to the service of the gateway, are not the ones from the service yaml block but the global ones. This PR addresses that issue.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [X] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Review by morvencao — APPROVED** [review]

File: manifests/charts/gateway/templates/service.yaml [hunk 1]
```diff
@@ -10,6 +10,7 @@ metadata:
     topology.istio.io/network: "{{.}}"
     {{- end }}
   annotations:
+    {{- .Values.service.annotations | toYaml | nindent 4 }}
```
Comment (Monkeyanator) [inline manifests/charts/gateway/templates/service.yaml:13]: Think we can fix tests by making this:
```yaml
  annotations:
    {{- with .Values.service.annotations }}
      {{- . | toYaml | nindent 4 }}
    {{- end }}
    {{- with .Values.annotations }}
      {{- . | toYaml | nindent 4 }}
    {{- end }}
```
not sure why it doesn't work as is...

**Comment by 4c74356b41** [comment]
https://github.com/istio/istio/pull/36384 was merged ahead of this

**Comment by howardjohn** [comment]
Your PR is appreciated! Since this was merged in https://github.com/istio/istio/pull/36384 I will close this. Thanks!

**Pull request closed without merging**